### ISSUE TITLE
Problem with generated types

### ISSUE THREAD

**Issue by fjbulic** [kubb-project/kubb#892]
### What version of `kubb` is running?

2.11.1

### What platform is your computer?

MacOS

### What version of external packages are you using(`@tanstack-query`, `MSW`, `React`, `Vue`, ...)

```
{
"dependencies": {
		"@chakra-ui/cli": "2.4.1",
		"@chakra-ui/next-js": "2.2.0",
		"@chakra-ui/react": "2.8.2",
		"@emotion/react": "11.11.4",
		"@emotion/styled": "11.11.0",
		"@hookform/error-message": "2.0.1",
		"@sentry/nextjs": "7.107.0",
		"@tanstack/react-query": "5.25.0",
		"@tanstack/react-query-devtools": "5.25.0",
		"axios": "1.6.8",
		"date-fns": "3.6.0",
		"decimal.js": "10.4.3",
		"express": "4.18.3",
		"framer-motion": "11.0.14",
		"http-proxy-middleware": "2.0.6",
		"next": "14.1.3",
		"next-safe": "3.5.0",
		"next-translate": "2.6.2",
		"next-translate-plugin": "2.6.2",
		"next-usequerystate": "1.17.1",
		"prom-client": "15.1.0",
		"react": "18.2.0",
		"react-dom": "18.2.0",
		"react-hook-form": "7.51.1",
		"sharp": "0.32.5",
		"zod": "3.22.4",
		"zustand": "4.5.2"
	},
	"devDependencies": {
		"@chakra-ui/cli": "2.4.1",
		"@kubb/cli": "2.11.1",
		"@kubb/core": "2.11.1",
		"@kubb/swagger": "2.11.1",
		"@kubb/swagger-tanstack-query": "2.11.1",
		"@kubb/swagger-ts": "2.11.1",
		"@prop/eslint-config": "workspace:*",
		"@prop/ts-config": "workspace:*",
		"@rushstack/eslint-patch": "1.3.3",
		"@types/fs-extra": "11.0.4",
		"@types/node": "18.15.12",
		"@types/prettier": "2.7.3",
		"@types/react": "18.2.66",
		"@types/react-dom": "18.2.22",
		"dedent": "1.5.1",
		"eslint": "8.48.0",
		"eslint-plugin-chakra-ui": "0.11.0",
		"fs-extra": "11.2.0",
		"glob": "10.3.10",
		"next-router-mock": "0.9.12",
		"node-html-parser": "6.1.12",
		"prettier": "2.8.8",
		"ts-node": "10.9.2",
		"typescript": "5.3.3"
	},
}
```

### What steps can reproduce the bug?

Seems like it's project specific, but can't seem to get to the bottom of it:

```ts
import client from '@kubb/swagger-client/client';
import {
	useQuery,
	queryOptions,
	useInfiniteQuery,
	infiniteQueryOptions,
	useSuspenseQuery,
} from '@tanstack/react-query';
import type { AffiliatesControllerAffiliateMeQueryResponse } from '../models/AffiliatesControllerAffiliateMe';
import type {
	QueryObserverOptions,
	UseQueryResult,
	QueryKey,
	InfiniteQueryObserverOptions,
	UseInfiniteQueryResult,
	InfiniteData,
	UseSuspenseQueryOptions,
	UseSuspenseQueryResult,
} from '@tanstack/react-query';

type AffiliatesControllerAffiliateMeClient = typeof client<
	AffiliatesControllerAffiliateMeQueryResponse,
	never,
	never
>;
type AffiliatesControllerAffiliateMe = {
	data: AffiliatesControllerAffiliateMeQueryResponse;
	error: never;
	request: never;
	pathParams: never;
	queryParams: never;
	headerParams: never;
	response: AffiliatesControllerAffiliateMeQueryResponse;
	client: {
		parameters: Partial<Parameters<AffiliatesControllerAffiliateMeClient>[0]>;
		return: Awaited<ReturnType<AffiliatesControllerAffiliateMeClient>>;
	};
};
export const affiliatesControllerAffiliateMeQueryKey = () => [{ url: '/affiliates/me' }] as const;
export type AffiliatesControllerAffiliateMeQueryKey = ReturnType<
	typeof affiliatesControllerAffiliateMeQueryKey
>;
export function affiliatesControllerAffiliateMeQueryOptions(
	options: AffiliatesControllerAffiliateMe['client']['parameters'] = {}
) {
	const queryKey = affiliatesControllerAffiliateMeQueryKey();
	return queryOptions({
		queryKey,
		queryFn: async () => {
			const res = await client<
				AffiliatesControllerAffiliateMe['data'],
				AffiliatesControllerAffiliateMe['error']
			>({
				method: 'get',
				url: `/affiliates/me`,
				...options,
			});
			return res.data;
		},
	});
}
/**
 * @description Get current user affiliate data, returns 4xx is user is not affiliate
 * @link /affiliates/me
 */
export function useAffiliatesControllerAffiliateMe<
	TData = AffiliatesControllerAffiliateMe['response'],
	TQueryData = AffiliatesControllerAffiliateMe['response'],
	TQueryKey extends QueryKey = AffiliatesControllerAffiliateMeQueryKey
>(
	options: {
		query?: Partial<
			QueryObserverOptions<
				AffiliatesControllerAffiliateMe['response'],
				AffiliatesControllerAffiliateMe['error'],
				TData,
				TQueryData,
				TQueryKey
			>
		>;
		client?: AffiliatesControllerAffiliateMe['client']['parameters'];
	} = {}
): UseQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? affiliatesControllerAffiliateMeQueryKey();
	const query = useQuery({
		...(affiliatesControllerAffiliateMeQueryOptions(clientOptions) as QueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<QueryObserverOptions, 'queryKey'>),
	}) as UseQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}
export const affiliatesControllerAffiliateMeInfiniteQueryKey = () =>
	[{ url: '/affiliates/me' }] as const;
export type AffiliatesControllerAffiliateMeInfiniteQueryKey = ReturnType<
	typeof affiliatesControllerAffiliateMeInfiniteQueryKey
>;
export function affiliatesControllerAffiliateMeInfiniteQueryOptions(
	options: AffiliatesControllerAffiliateMe['client']['parameters'] = {}
) {
	const queryKey = affiliatesControllerAffiliateMeInfiniteQueryKey();
	return infiniteQueryOptions({
		queryKey,
		queryFn: async ({ pageParam }) => {
			const res = await client<
				AffiliatesControllerAffiliateMe['data'],
				AffiliatesControllerAffiliateMe['error']
			>({
				method: 'get',
				url: `/affiliates/me`,
				...options,
			});
			return res.data;
		},
		initialPageParam: 0,
		getNextPageParam: (lastPage, allPages, lastPageParam) =>
			Array.isArray(lastPage) && lastPage.length === 0 ? undefined : lastPageParam + 1,
		getPreviousPageParam: (firstPage, allPages, firstPageParam) =>
			firstPageParam <= 1 ? undefined : firstPageParam - 1,
	});
}
/**
 * @description Get current user affiliate data, returns 4xx is user is not affiliate
 * @link /affiliates/me
 */
export function useAffiliatesControllerAffiliateMeInfinite<
	TData = InfiniteData<AffiliatesControllerAffiliateMe['response']>,
	TQueryData = AffiliatesControllerAffiliateMe['response'],
	TQueryKey extends QueryKey = AffiliatesControllerAffiliateMeInfiniteQueryKey
>(
	options: {
		query?: Partial<
			InfiniteQueryObserverOptions<
				AffiliatesControllerAffiliateMe['response'],
				AffiliatesControllerAffiliateMe['error'],
				TData,
				TQueryData,
				TQueryKey
			>
		>;
		client?: AffiliatesControllerAffiliateMe['client']['parameters'];
	} = {}
): UseInfiniteQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? affiliatesControllerAffiliateMeInfiniteQueryKey();
	const query = useInfiniteQuery({
		...(affiliatesControllerAffiliateMeInfiniteQueryOptions(
			clientOptions
		) as InfiniteQueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<InfiniteQueryObserverOptions, 'queryKey'>),
	}) as UseInfiniteQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}
export const affiliatesControllerAffiliateMeSuspenseQueryKey = () =>
	[{ url: '/affiliates/me' }] as const;
export type AffiliatesControllerAffiliateMeSuspenseQueryKey = ReturnType<
	typeof affiliatesControllerAffiliateMeSuspenseQueryKey
>;
export function affiliatesControllerAffiliateMeSuspenseQueryOptions(
	options: AffiliatesControllerAffiliateMe['client']['parameters'] = {}
) {
	const queryKey = affiliatesControllerAffiliateMeSuspenseQueryKey();
	return queryOptions({
		queryKey,
		queryFn: async () => {
			const res = await client<
				AffiliatesControllerAffiliateMe['data'],
				AffiliatesControllerAffiliateMe['error']
			>({
				method: 'get',
				url: `/affiliates/me`,
				...options,
			});
			return res.data;
		},
	});
}
/**
 * @description Get current user affiliate data, returns 4xx is user is not affiliate
 * @link /affiliates/me
 */
export function useAffiliatesControllerAffiliateMeSuspense<
	TData = AffiliatesControllerAffiliateMe['response'],
	TQueryKey extends QueryKey = AffiliatesControllerAffiliateMeSuspenseQueryKey
>(
	options: {
		query?: Partial<
			UseSuspenseQueryOptions<
				AffiliatesControllerAffiliateMe['response'],
				AffiliatesControllerAffiliateMe['error'],
				TData,
				TQueryKey
			>
		>;
		client?: AffiliatesControllerAffiliateMe['client']['parameters'];
	} = {}
): UseSuspenseQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? affiliatesControllerAffiliateMeSuspenseQueryKey();
	const query = useSuspenseQuery({
		...(affiliatesControllerAffiliateMeSuspenseQueryOptions(clientOptions) as QueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<QueryObserverOptions, 'queryKey'>),
	}) as UseSuspenseQueryResult<TData, AffiliatesControllerAffiliateMe['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}

```

Typescript errors on the `as QueryObserverOptions` castings in the files:

```shell
Conversion of type 'UseQueryOptions<AffiliateDto, Error, AffiliateDto, readonly [{ readonly url: "/affiliates/me"; }]> & { initialData?: undefined; } & { ...; }' to type 'QueryObserverOptions<unknown, Error, unknown, unknown, QueryKey, never>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'persister' are incompatible.
    Type '(queryFn: QueryFunction<AffiliateDto, readonly [{ readonly url: "/affiliates/me"; }], never>, context: { queryKey: readonly [{ readonly url: "/affiliates/me"; }]; signal: AbortSignal; meta: Record<...> | undefined; }, query: Query<...>) => AffiliateDto | Promise<...>' is not comparable to type '(queryFn: QueryFunction<unknown, QueryKey, never>, context: { queryKey: QueryKey; signal: AbortSignal; meta: Record<string, unknown> | undefined; }, query: Query<...>) => unknown'.
      Types of parameters 'context' and 'context' are incompatible.
        Type '{ queryKey: QueryKey; signal: AbortSignal; meta: Record<string, unknown> | undefined; }' is not comparable to type '{ queryKey: readonly [{ readonly url: "/affiliates/me"; }]; signal: AbortSignal; meta: Record<string, unknown> | undefined; }'.
          Types of property 'queryKey' are incompatible.
            Type 'QueryKey' is not comparable to type 'readonly [{ readonly url: "/affiliates/me"; }]'.
              Target requires 1 element(s) but source may have fewer.
```

since query keys generated by query options functions is not compatible with `queryOptions?.queryKey`. Any ideas?

### How often does this bug happen?

Every time

### What is the expected behavior?

No errors with generated hooks.

**Comment by fjbulic** [comment]
Tried updating the dependencies:

```
	"dependencies": {
		"@chakra-ui/cli": "2.4.1",
		"@chakra-ui/next-js": "2.2.0",
		"@chakra-ui/react": "2.8.2",
		"@emotion/react": "11.11.4",
		"@emotion/styled": "11.11.0",
		"@hookform/error-message": "2.0.1",
		"@sentry/nextjs": "7.107.0",
		"@tanstack/react-query": "5.28.9",
		"@tanstack/react-query-devtools": "5.28.9",
		"axios": "1.6.8",
		"date-fns": "3.6.0",
		"framer-motion": "11.0.14",
		"http-proxy-middleware": "2.0.6",
		"next": "14.1.3",
		"next-safe": "3.5.0",
		"next-translate": "2.6.2",
		"next-translate-plugin": "2.6.2",
		"next-usequerystate": "1.17.1",
		"react": "18.2.0",
		"react-dom": "18.2.0",
		"react-hook-form": "7.51.1",
		"sharp": "0.32.5",
		"zustand": "4.5.2"
	},
	"devDependencies": {
		"@chakra-ui/cli": "2.4.1",
		"@kubb/cli": "2.7.0",
		"@kubb/core": "2.7.0",
		"@kubb/swagger": "2.7.0",
		"@kubb/swagger-tanstack-query": "2.7.0",
		"@kubb/swagger-ts": "2.7.0",
		"@prop/eslint-config": "workspace:*",
		"@prop/ts-config": "workspace:*",
		"@rushstack/eslint-patch": "1.3.3",
		"@types/fs-extra": "11.0.4",
		"@types/node": "18.15.12",
		"@types/react": "18.2.66",
		"@types/react-dom": "18.2.22",
		"dedent": "1.5.1",
		"eslint": "8.48.0",
		"eslint-plugin-chakra-ui": "0.11.0",
		"fs-extra": "11.2.0",
		"glob": "10.3.10",
		"node-html-parser": "6.1.12",
		"prettier": "2.8.8",
		"typescript": "5.3.3"
	},
```

and generated hooks for https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/api-with-examples.json. But still the same errors:

```tsx
import client from '@utils/client';
import {
	useQuery,
	queryOptions,
	useInfiniteQuery,
	infiniteQueryOptions,
	useSuspenseQuery,
} from '@tanstack/react-query';
import type { ListVersionsv2QueryResponse } from '../models/ListVersionsv2';
import type {
	QueryObserverOptions,
	UseQueryResult,
	QueryKey,
	InfiniteQueryObserverOptions,
	UseInfiniteQueryResult,
	InfiniteData,
	UseSuspenseQueryOptions,
	UseSuspenseQueryResult,
} from '@tanstack/react-query';

type ListVersionsv2Client = typeof client<ListVersionsv2QueryResponse, never, never>;
type ListVersionsv2 = {
	data: ListVersionsv2QueryResponse;
	error: never;
	request: never;
	pathParams: never;
	queryParams: never;
	headerParams: never;
	response: ListVersionsv2QueryResponse;
	client: {
		parameters: Partial<Parameters<ListVersionsv2Client>[0]>;
		return: Awaited<ReturnType<ListVersionsv2Client>>;
	};
};
export const listVersionsv2QueryKey = () => [{ url: '/' }] as const;
export type ListVersionsv2QueryKey = ReturnType<typeof listVersionsv2QueryKey>;
export function listVersionsv2QueryOptions(options: ListVersionsv2['client']['parameters'] = {}) {
	const queryKey = listVersionsv2QueryKey();
	return queryOptions({
		queryKey,
		queryFn: async () => {
			const res = await client<ListVersionsv2['data'], ListVersionsv2['error']>({
				method: 'get',
				url: `/`,
				...options,
			});
			return res.data;
		},
	});
}
/**
 * @summary List API versions
 * @link / */
export function useListVersionsv2<
	TData = ListVersionsv2['response'],
	TQueryData = ListVersionsv2['response'],
	TQueryKey extends QueryKey = ListVersionsv2QueryKey
>(
	options: {
		query?: Partial<
			QueryObserverOptions<
				ListVersionsv2['response'],
				ListVersionsv2['error'],
				TData,
				TQueryData,
				TQueryKey
			>
		>;
		client?: ListVersionsv2['client']['parameters'];
	} = {}
): UseQueryResult<TData, ListVersionsv2['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? listVersionsv2QueryKey();
	const query = useQuery({
		...(listVersionsv2QueryOptions(clientOptions) as QueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<QueryObserverOptions, 'queryKey'>),
	}) as UseQueryResult<TData, ListVersionsv2['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}
export const listVersionsv2InfiniteQueryKey = () => [{ url: '/' }] as const;
export type ListVersionsv2InfiniteQueryKey = ReturnType<typeof listVersionsv2InfiniteQueryKey>;
export function listVersionsv2InfiniteQueryOptions(
	options: ListVersionsv2['client']['parameters'] = {}
) {
	const queryKey = listVersionsv2InfiniteQueryKey();
	return infiniteQueryOptions({
		queryKey,
		queryFn: async ({ pageParam }) => {
			const res = await client<ListVersionsv2['data'], ListVersionsv2['error']>({
				method: 'get',
				url: `/`,
				...options,
			});
			return res.data;
		},
		initialPageParam: 0,
		getNextPageParam: (lastPage, allPages, lastPageParam) =>
			Array.isArray(lastPage) && lastPage.length === 0 ? undefined : lastPageParam + 1,
		getPreviousPageParam: (firstPage, allPages, firstPageParam) =>
			firstPageParam <= 1 ? undefined : firstPageParam - 1,
	});
}
/**
 * @summary List API versions
 * @link / */
export function useListVersionsv2Infinite<
	TData = InfiniteData<ListVersionsv2['response']>,
	TQueryData = ListVersionsv2['response'],
	TQueryKey extends QueryKey = ListVersionsv2InfiniteQueryKey
>(
	options: {
		query?: Partial<
			InfiniteQueryObserverOptions<
				ListVersionsv2['response'],
				ListVersionsv2['error'],
				TData,
				TQueryData,
				TQueryKey
			>
		>;
		client?: ListVersionsv2['client']['parameters'];
	} = {}
): UseInfiniteQueryResult<TData, ListVersionsv2['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? listVersionsv2InfiniteQueryKey();
	const query = useInfiniteQuery({
		...(listVersionsv2InfiniteQueryOptions(clientOptions) as InfiniteQueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<InfiniteQueryObserverOptions, 'queryKey'>),
	}) as UseInfiniteQueryResult<TData, ListVersionsv2['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}
export const listVersionsv2SuspenseQueryKey = () => [{ url: '/' }] as const;
export type ListVersionsv2SuspenseQueryKey = ReturnType<typeof listVersionsv2SuspenseQueryKey>;
export function listVersionsv2SuspenseQueryOptions(
	options: ListVersionsv2['client']['parameters'] = {}
) {
	const queryKey = listVersionsv2SuspenseQueryKey();
	return queryOptions({
		queryKey,
		queryFn: async () => {
			const res = await client<ListVersionsv2['data'], ListVersionsv2['error']>({
				method: 'get',
				url: `/`,
				...options,
			});
			return res.data;
		},
	});
}
/**
 * @summary List API versions
 * @link / */
export function useListVersionsv2Suspense<
	TData = ListVersionsv2['response'],
	TQueryKey extends QueryKey = ListVersionsv2SuspenseQueryKey
>(
	options: {
		query?: Partial<
			UseSuspenseQueryOptions<ListVersionsv2['response'], ListVersionsv2['error'], TData, TQueryKey>
		>;
		client?: ListVersionsv2['client']['parameters'];
	} = {}
): UseSuspenseQueryResult<TData, ListVersionsv2['error']> & {
	queryKey: TQueryKey;
} {
	const { query: queryOptions, client: clientOptions = {} } = options ?? {};
	const queryKey = queryOptions?.queryKey ?? listVersionsv2SuspenseQueryKey();
	const query = useSuspenseQuery({
		...(listVersionsv2SuspenseQueryOptions(clientOptions) as QueryObserverOptions),
		queryKey,
		...(queryOptions as unknown as Omit<QueryObserverOptions, 'queryKey'>),
	}) as UseSuspenseQueryResult<TData, ListVersionsv2['error']> & {
		queryKey: TQueryKey;
	};
	query.queryKey = queryKey as TQueryKey;
	return query;
}

```

every `as QueryObserverOptions` cast errros with something like:

```
Conversion of type 'UseQueryOptions<unknown, Error, unknown, readonly [{ readonly url: "/"; }]> & { initialData?: undefined; } & { queryKey: DataTag<readonly [{ readonly url: "/"; }], unknown>; }' to type 'QueryObserverOptions<unknown, Error, unknown, unknown, QueryKey, never>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'persister' are incompatible.
    Type '(queryFn: QueryFunction<unknown, readonly [{ readonly url: "/"; }], never>, context: { queryKey: readonly [{ readonly url: "/"; }]; signal: AbortSignal; meta: Record<string, unknown> | undefined; }, query: Query<...>) => unknown' is not comparable to type '(queryFn: QueryFunction<unknown, QueryKey, never>, context: { queryKey: QueryKey; signal: AbortSignal; meta: Record<string, unknown> | undefined; }, query: Query<...>) => unknown'.
      Types of parameters 'context' and 'context' are incompatible.
        Type '{ queryKey: QueryKey; signal: AbortSignal; meta: Record<string, unknown> | undefined; }' is not comparable to type '{ queryKey: readonly [{ readonly url: "/"; }]; signal: AbortSignal; meta: Record<string, unknown> | undefined; }'.
          Types of property 'queryKey' are incompatible.
            Type 'QueryKey' is not comparable to type 'readonly [{ readonly url: "/"; }]'.
              Target requires 1 element(s) but source may have fewer.ts(2352)
```

**Comment by fjbulic** [comment]
Here is my tsconfig:

```
{
	"$schema": "https://json.schemastore.org/tsconfig",
	"compilerOptions": {
		"strict": true,
		"allowUnreachableCode": false,
		"allowUnusedLabels": false,
		"exactOptionalPropertyTypes": true,
		"noFallthroughCasesInSwitch": true,
		"noImplicitOverride": true,
		"noImplicitReturns": true,
		"noPropertyAccessFromIndexSignature": true,
		"noUncheckedIndexedAccess": true,
		"noUnusedLocals": true,
		"noUnusedParameters": true,

		"module": "ESNext",
		"moduleResolution": "Bundler",
		"resolveJsonModule": true,

		"declaration": true,
		"declarationMap": true,
		"removeComments": true,

		"allowJs": false,

		"esModuleInterop": true,
		"isolatedModules": true,
		"forceConsistentCasingInFileNames": true,

		"target": "ESNext",
		"emitDecoratorMetadata": true,
		"experimentalDecorators": true,

		"composite": false,
		"incremental": true,

		"preserveWatchOutput": true,

		"skipLibCheck": true,

		"assumeChangesOnlyAffectDirectDependencies": true
	}
}

```

the error is gone when disabling `exactOptionalPropertyTypes`, but would appreciate if it worked with the flag enabled 🙏

**Comment by stijnvanhulle** [comment]
Fixed with version 2.12.1
